src/gluonts/shell/sagemaker/__init__.py

```diff
@@ -131,8 +134,19 @@ def _load_datasets(
     hyperparameters: dict, channels: Dict[str, Path]
 ) -> Dict[str, FileDataset]:
     freq = hyperparameters["freq"]
-    return {
-        name: FileDataset(channels[name], freq)
-        for name in DATASET_NAMES
-        if name in channels
-    }
+    listify_dataset = (
+        strtobool(hyperparameters["listify_dataset"])
+        if "listify_dataset" in hyperparameters.keys()
+        else False
+    )
+    dataset_dict = {}
+    for name in DATASET_NAMES:
+        if name in channels:
+            file_dataset = FileDataset(channels[name], freq)
+            dataset_dict[name] = (
+                ListDataset(file_dataset, freq)
+                if listify_dataset
+                else file_dataset
+            )
```

Comment:
I would add the try catch logic for catching OOM due to the conversion of FileDataset to ListDataset here.